[docs] review and update bare installation guide for dev-client (#19295)
expo/expo: docs/pages/development/installation.md

<Terminal cmd={["$ npx crna -t with-dev-client"]} />

Otherwise, follow this guide to add `expo-dev-client` to your existing React Native or Bare project.

> ## 0. Set up Expo modules

Reviewer: ```suggestion
## 1. Set up Expo modules
```

As per our [writing style guide here](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md), for consistency reasons, let's start numbered lists with `1` 😄

If you created your project with `npx react-native init` and do not have any other Expo packages installed, **you'll need to [install Expo modules](../bare/installing-expo-modules)** before proceeding.

If you created your project with `expo init` or `create-expo-app`, or you already have `expo` and/or other Expo modules up and running, you can skip this step.

> ## 1. Installation

Reviewer: ```suggestion
## 2. Installation
```

Add the `expo-dev-client` library to your package.json.

<InstallSection packageName="expo-development-client" cmd={["npx expo install expo-dev-client"]} hideBareInstructions />

<Tabs tabs={["SDK 45+", "SDK below 45"]}>

<Tab >

> No additional changes are needed to install the package on SDK 45+.

Reviewer: ```suggestion
No additional changes are needed to install the package on SDK 45 and above.
```

As per the [writing style guide here](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#symbols-as-words), let's avoid using symbols.

<img src="/static/images/client/check_ios_version.png" style={{maxWidth: "100%" }}/>

### Android

No additional changes are needed to install the package on Android.

</Tab >

</Tabs >

> ## 2. Basic configuration

Reviewer: ```suggestion
## 3. Basic configuration
```

Also, can I please ask you to change the numbered list pattern for the rest of the sections? 🙏 

I'd have done it myself but I can't review the unchanged diff 😄

<Tabs tabs={["SDK 45+", "SDK below 45"]}>

<Tab >

> If you already have your `ios` or `android` directories on disk, you will need to rerun the config plugin in order to configure `expo-dev-client`:

Reviewer: ```suggestion
If you already have your `ios` or `android` directories on disk, you will need to re-run the config plugin to configure `expo-dev-client`:
```

<Terminal cmd={["$ npx expo prebuild"]} />

If you don't have an `ios` or `android` directory, you can skip this step.

</Tab >

<Tab >

Development builds use deep links to open projects from the QR code. If you have added a custom deep link scheme to your project, your development build will use it. However, if this isn't the case, you need to configure the deep link support for your app. The `uri-scheme` package will do this for you once you have chosen a scheme.

<InstallSection packageName="expo-development-client" cmd={["npx uri-scheme add <your scheme>"]} hideBareInstructions />

> See the [uri-scheme package](https://www.npmjs.com/package/uri-scheme) for more information.

Reviewer: ```suggestion
For more information, see [uri-scheme package](https://www.npmjs.com/package/uri-scheme).
```

There are a few more changes you can make to get the best experience, but you [can skip ahead to building](/development/getting-started/#building-and-installing-your-first-development-build) if you prefer.

### Disable packager autostart when building for iOS

> When you start your project on iOS, the metro bundler will be started automatically. This behavior might not be ideal when you want to use `npx expo start`. Our recommended solution is to remove the `Start Packager` action from building scripts. To do that you need to open the Xcode, go to "Build Phases" and remove the "Start Packager" action.

Reviewer: ```suggestion
When you start your project on iOS, the metro bundler will start automatically. This behavior might not be ideal when you want to use `npx expo start`. Our recommended solution is to remove the `Start Packager` action from building scripts. To do that you need to open the Xcode, go to "Build Phases" and remove the "Start Packager" action.
```

These suggestions are unrelated to this PR but I think it's a good improvement.

### Loading published updates

Development builds can also open and preview published updates to your app.

<Tabs tabs={["SDK 45+", "SDK below 45"]}>

<Tab>

To add this feature, install `expo-updates` in your project:

<Terminal cmd={["$ npx expo install expo-updates"]} />

> You may need to rerun the config plugin for `expo-dev-client` after doing so.

Reviewer: ```suggestion
After doing so, you may need to re-run the config plugin for `expo-dev-client`.
```

I don't get the context of why this step is a "may"? Is it like some users have to and some don't if they install `expo-updates`?

Author: Thanks for making me think about this a little harder -- I realized that this step isn't actually necessary at all! 😅  Rewrote a few other places to reflect this, as well.

Reviewer: Thanks for making these changes 🙏 
Love the new change "SDK 44 and below". I think it is more clear to imply the info docs are conveying there.
